qiskit/dagcircuit/dagnode.py

```diff
@@ -113,16 +115,20 @@ def __str__(self):
         return str(id(self))
 
     @staticmethod
-    def semantic_eq(node1, node2):
+    def semantic_eq(node1: 'DAGNode', node2: 'DAGNode') -> bool:
```

Comment:
what are the single quotations around the type here for?

Reply:
Since the type hint was referring to the type of the class itself, I needed to make the type hint a string. Otherwise, I got an error from the linter I believe.